molecule/default/tasks/generate_name.yml

```diff
@@ -0,0 +1,188 @@
+    assert:
+      that:
+      - result is failed
+      - "result.msg == 'At least one of metadata.name|metadata.generateName is required to create object.'"
+
+  - name: create pod using name parameter should succeed
+    k8s:
+      namespace: '{{ namespace }}'
+      definition: '{{ pod_00 }}'
+      name: pod-01
+
+  - name: list Pod for namespace
+    k8s_info:
+      kind: Pod
+      namespace: '{{ namespace }}'
+    register: pods
+
+  - name: assert pod has been created
+    assert:
+      that:
+      - '{{ pods.resources | length == 1 }}'
+
+  - name: create pod using generate_name parameter should succeed
+    k8s:
+      namespace: '{{ namespace }}'
+      definition: '{{ pod_00 }}'
+      generate_name: pod-
+
+  - name: list Pod for namespace
+    k8s_info:
+      kind: Pod
+      namespace: '{{ namespace }}'
+    register: pods
+
+  - name: assert pod has been created
+    assert:
+      that:
+      - '{{ pods.resources | length == 2 }}'
+
+  - name: create pod using metadata.generateName parameter should succeed
+    k8s:
+      namespace: '{{ namespace }}'
+      definition: '{{ pod_01 }}'
+
+  - name: list Pod for namespace
+    k8s_info:
+      kind: Pod
+      namespace: '{{ namespace }}'
+    register: pods
+
+  - name: assert pod has been created
+    assert:
+      that:
+      - '{{ pods.resources | length == 3 }}'
+
+  - name: create object using metadata.generateName should support wait option
+    k8s:
+      namespace: '{{ namespace }}'
+      definition:
+        apiVersion: apps/v1
+        kind: StatefulSet
+        metadata:
+          generateName: test-
+        spec:
+          selector:
+            matchLabels:
+              app: nginx
+          serviceName: "nginx"
+          replicas: 3
+          template:
+            metadata:
+              labels:
+                app: nginx
+            spec:
+              terminationGracePeriodSeconds: 10
+              containers:
+              - name: nginx
+                image: k8s.gcr.io/nginx-slim:0.8
+                ports:
+                - containerPort: 80
+                  name: web
+      wait: yes
+      wait_sleep: 3
+      wait_timeout: 180
+
+  - name: Create ConfigMap using generateName
+    kubernetes.core.k8s:
+      kind: ConfigMap
+      namespace: '{{ namespace }}'
+      generate_name: cmap-
+      append_hash: yes
+    register: config
+
+  - name: assert that configmap has been created using generateName
+    assert:
+      that:
+      - "config.result.metadata.name.startswith('cmap-')"
+
+  - name: Create Pod with failing container
+    kubernetes.core.k8s:
+      namespace: '{{ namespace }}'
+      definition:
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: pod1
+        spec:
+          containers:
+          - image: adslfkjadslfkjadslkfjsadf
+            name: non-existent-container-image
+
+  - name: Create second Pod using wait (it should not wait for the first container)
```

Comment:
The important bit for this one is that the pod definition should *not* include a `name`, but you should use the `generate_name` param, instead. This does appear to be passing locally for me now with the new changes, but we should update the test.

Reply:
fixed.
Thanks for the feedback